ts/main.ts

```diff
@@ -83,44 +143,54 @@ function swapView(anime: Anime): void {
   const img = document.createElement('img');
   img.src = anime.images.jpg.image_url;
   img.alt = anime.title;
-
-  // Clear existing children in imageElement and append new image
-  while (imageElement.firstChild) {
-    imageElement.removeChild(imageElement.firstChild);
-  }
   imageElement.appendChild(img);
 
   // Update rating
   ratingElement.textContent = `Rating: ${anime.rating ?? 'N/A'}`;
 
+  // Add or Remove from Favorites button
+  const isFavorite = favorites.some((fav) => fav.mal_id === anime.mal_id);
+  const starButton = createStarButton(isFavorite);
+  starButton.addEventListener('click', () => {
+    updateFavorites(anime);
+    const isFavoriteNow = favorites.some((fav) => fav.mal_id === anime.mal_id);
+    starButton
+      .querySelector('path')!
+      .setAttribute('fill', isFavoriteNow ? 'yellow' : 'none');
+  });
+
+  // Append the star button
+  const detailRight = detailView.querySelector(
+    '.detail-right',
+  ) as HTMLDivElement;
+  detailRight.appendChild(starButton);
+
   // Show detail view and hide anime container
   animeContainer.classList.add('hidden');
   detailView.classList.remove('hidden');
 }
 
-// starts the page
-async function init(): Promise<void> {
-  const animeList = await fetchAnime();
-
+// Render the anime list
+function renderList(animeList: Anime[], container: HTMLDivElement): void {
+  container.innerHTML = ''; // Clear existing content
   animeList.forEach((anime) => {
-    animeContainer.appendChild(renderAnime(anime));
+    container.appendChild(renderAnime(anime));
   });
+}
 
-  // Event delegation
-  animeContainer.addEventListener('click', (event: Event) => {
-    const target = event.target as HTMLElement;
-    const animeDiv = target.closest('.animes') as HTMLDivElement;
+// Initialize the app
+async function init(): Promise<void> {
+  await fetchAnime(); // Fetch anime data and update local storage
 
-    if (animeDiv) {
-      const animeId = animeDiv.dataset.id;
-      const selectedAnime = animeList.find(
-        (a) => a.mal_id.toString() === animeId,
-      );
+  // Event for favorites link (not implemented yet)
+  favoritesLink.addEventListener('click', () => {
+    // Placeholder for future functionality to switch to favorites view
+  });
 
-      if (selectedAnime) {
-        swapView(selectedAnime);
-      }
-    }
+  // Event for back to main page
+  document.querySelector('.home-page')?.addEventListener('click', () => {
+    detailView.classList.add('hidden');
+    animeContainer.classList.remove('hidden');
```

Comment:
We would want to ensure this is _not_ `null` rather than using the optional chaining operator.